### PR TITLE
support specify audio encode bitrateMode interface

### LINKED ISSUE #649: do we need support specify audio encode bitrateMode interface ?
some audio codecs(opus, HE AAC) support specify  encoder bitrate mode, 

according [section-2.1.8 of opus rfc6716](https://datatracker.ietf.org/doc/html/rfc6716#section-2.1.8) and [opus api doc](https://www.opus-codec.org/docs/html_api/group__encoderctls.html#ga34d09ae06cab7e1a6c49876249b67892) , 
opus encoder support VBR and CBR mode, and vbr was used by default.

we can use [bitrateMode exist in mediastream-recording](https://www.w3.org/TR/mediastream-recording/#enumdef-bitratemode), 
`constant` correspond to CBR and `variable` correspond to VBR mode. 

suggest to add `audioBitrateMode` in audio codec that support vbr/cbr mode,   which specifes the [BitrateMode](https://www.w3.org/TR/mediastream-recording/#enumdef-bitratemode) that should be used to encode the audio track.

### PR DESCRIPTION
**Pull request by bdrtc**
fixes #649

***
<a href="https://pr-preview.s3.amazonaws.com/bdrtc/webcodecs/pull/657.html" title="Last updated on Apr 18, 2023, 9:18 AM UTC (e54743a)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webcodecs/657/2b856eb...bdrtc:e54743a.html" title="Last updated on Apr 18, 2023, 9:18 AM UTC (e54743a)">Diff</a>

### CODE REVIEW

**Comment by dalecurtis** [comment]
Do we want a config level bitrate mode knob like we have for video instead?

**Comment by Djuffin** [comment]
I agree with @dalecurtis, bitrate mode seems like a very generic knob that will be used for many codecs with the same meaning. 
So I think it belongs in AudioEncoderConfig

**Comment by bdrtc** [comment]
> AudioEncoderConfig

Thanks feedback, I  considered add this to AudioEncoderConfig before commit, but for audio codec, not all codecs support vbr mode, For example ulaw/alaw, what will happen if user specify a bitrate mode to AudioEncoderConfig that current codec does't supported ?

**Comment by Djuffin** [comment]
> what will happen if user specify a bitrate mode to AudioEncoderConfig that current codec does't supported ?

[AudioEncoder.isConfigSupported()](https://www.w3.org/TR/webcodecs/#dom-audioencoder-isconfigsupported) will return false.

**Comment by bdrtc** [comment]
> Do we want a config level bitrate mode knob like we have for video instead?

I will move it to `AudioEncoderConfig` and add a Note, Thanks.

**Comment by bdrtc** [comment]
Dear editors, any more comments about this ? thanks.

**Comment by Djuffin** [comment]
LGTM @tguilbert-google please take a look

File: index.src.html [hunk 1]
```diff
@@ -2056,6 +2057,13 @@
   <dd>
     The average bitrate of the encoded audio given in units of bits per second.
   </dd>
+  <dt><dfn dict-member for=AudioEncoderConfig>bitrateMode</dfn></dt>
+  <dd>
+    Configures encoder to use a {{BitrateMode/constant}} or
```
Comment (chrisn) [inline index.src.html:2062]: Suggest "Configures the encoder"

File: index.src.html [hunk 2]
```diff
@@ -2056,6 +2057,13 @@
...
+    {{BitrateMode/variable}} bitrate as defined by [[MEDIASTREAM-RECORDING]].
+
+	NOTE: Not all audio codecs support specific {{BitrateMode}}, Authors are encouraged to check by calling {{AudioEncoder/isConfigSupported()}} with |config|.
```
Comment (chrisn) [inline index.src.html:2065]: Suggest "support specific {{BitrateMode}}s" or "support setting a specific {{BitrateMode}}"

**Review by chrisn — CHANGES_REQUESTED** [review]

**Comment by bdrtc** [comment]
> Also LGTM!

Thanks. If there's any questions please let me know when you approve this.

**Pull request merged**